gallifreyan.js

```diff
@@ -209,7 +226,15 @@ function doClick(e) {
             }
         }
     }
-    if (selectedLine != null) selectedCircle = null;    //if we've selected a line, let's unselect a circle
+    if (selectedLine != null) {
+        selectedCircle = null;    //if we've selected a line, let's unselect a circle
+        if (deleteLineMode) {
+            deleteLine(selectedLine);
+            selectedLine = null;
+        }
+    }
+    deleteLineMode = false;

```

Comment:
`deleteLineMode = false` is not in that block, so that when the user clicks on "delete line" and then somewhere into a canvas, the deletion is aborted. From my point of view this is more user-friendly than to insist on deleting something when the user clicks on "delete line".